packages/fluentui/react/src/components/Input/Input.tsx

```diff
@@ -185,13 +190,23 @@ class Input extends AutoControlledComponent<WithAsProp<InputProps>, InputState>
 
   handleIconOverrides = predefinedProps => ({
     onClick: (e: React.SyntheticEvent) => {
+      if (this.props.disabled) {
+        e.preventDefault();
+        return;
+      }
+
       this.handleOnClear(e);
       this.inputRef.current.focus();
       _.invoke(predefinedProps, 'onClick', e, this.props);
     }
```

Comment:
```suggestion
      if (!this.props.disabled) {
        this.handleOnClear(e);
        this.inputRef.current.focus();
      }
      
      _.invoke(predefinedProps, 'onClick', e, this.props);
    }
```

Can you please explain why we should `preventDefault()` in this case?